test/circuit.jl

```diff
@@ -2,17 +2,33 @@ using Snowflurry
 using Test
 
 @testset "Constructor: QuantumCircuit" begin
-
     c = QuantumCircuit(qubit_count = 1)
 
+    @test get_num_qubits(c) == 1
+    @test get_num_bits(c) == 1
     @test length(get_circuit_instructions(c)) == 0
 
-    @test_throws DomainError QuantumCircuit(qubit_count = 1, instructions = [sigma_x(5)])
+    c = QuantumCircuit(qubit_count = 6, bit_count = 2, instructions = [sigma_x(5)])
 
-    @test_throws AssertionError QuantumCircuit(qubit_count = 0, instructions = [sigma_x(5)])
+    @test get_num_qubits(c) == 6
+    @test get_num_bits(c) == 2
+    @test length(get_circuit_instructions(c)) == 1
 
-    @test_throws AssertionError QuantumCircuit(qubit_count = 0)
+    @test_throws DomainError QuantumCircuit(qubit_count = 1, instructions = [sigma_x(5)])
 
+    @test_throws AssertionError(
+        "$(:QuantumCircuit) constructor requires qubit_count>0. Received: 0",
```

Comment:
Are we sure we want to couple the test to the string in the error? I think it makes tests more brittle in the long run.

If we're worried about making sure we throw an error for a specific reason, we can make two circuits; a 'valid' circuit and a 'pathological' circuit; the pathological circuit can be a clone/duplicate of the valid circuit with just the invalidity added that we're testing against... WDYT?

Reply:
Hum, this is more "are we throwing the error we think we're throwing". It's happened a few times that we shadowed one error with another

Reply:
I think the "right" way would be to define our own custom exceptions, but not for this MR. See the last part of the following section: https://docs.julialang.org/en/v1/manual/control-flow/#Built-in-Exceptions.

Reply:
Ok I'll revert it to just checking the error type and add an issue for making our own custom error types

Reply:
We are still checking against the error string in a few places. I'm not concerned if we are moving to custom exceptions. We can convert to a custom exception if the test string changes.

@caffeine-storm Are you happy with merging as is?

Reply:
I'm happy to have this land.